contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+                _oethbRequired - _oethbBalance
+            );
+        }
+
+        uint256 _wethAmountSupplied;
+        uint256 _oethbAmountSupplied;
+        if (tokenId == 0) {
+            (
+                uint256 _mintedTokenId,
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.mint(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     *
+                     * Also sqrtPriceX96 is 0 because the pool is already created
+                     * non zero amount attempts to create a new instance of the pool
+                     */
+                    INonfungiblePositionManager.MintParams({
+                        token0: WETH,
+                        token1: OETHb,
+                        tickSpacing: tickSpacing,
+                        tickLower: lowerTick,
+                        tickUpper: upperTick,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
```

Comment:
After the first initial deposit, the NFT will be owned by the gauge contract, a redeposit isn't possible. I have added the CLGauge contract code for you to have a look.

```solidity
File: CLGauge.sol
    function deposit(uint256 tokenId) external override nonReentrant {
        require(nft.ownerOf(tokenId) == msg.sender, "NA");
        require(voter.isAlive(address(this)), "GK");
        (,, address _token0, address _token1, int24 _tickSpacing, int24 tickLower, int24 tickUpper,,,,,) =
            nft.positions(tokenId);
        require(token0 == _token0 && token1 == _token1 && tickSpacing == _tickSpacing, "PM");

        // trigger update on staked position so NFT will be in sync with the pool
        nft.collect(
            INonfungiblePositionManager.CollectParams({
                tokenId: tokenId,
                recipient: msg.sender,
                amount0Max: type(uint128).max,
                amount1Max: type(uint128).max
            })
        );

        nft.safeTransferFrom(msg.sender, address(this), tokenId);
```

Reply:
Agreed, this has been addressed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/881a617a121896804e173c9bacccb8c4880dc25e